### DISCUSSION TITLE
Usage of message_bus

### DISCUSSION THREAD

**Discussion by tunguskar** [ETLCPP/etl#792] [Q&A]
Hi,

I used now a little bit the message_bus bib and the other parts. What I want to achieve is the following:

What I have:

Connection which consists of:
- A physical Connection
- A protocol

On top of this `Connection` comes the `MessageHandler` in this case I thought it is the `message_bus`. 

Other Components in the system have functionality which should be triggered from the received message, but the components should not know about the Message Handler. I thought, that the MessageHandler composes at startup the relationship between the `Feature` and the `Service`. The Service in this case is the MessageHandler which has a raw Message on hand. 

In All the examples I saw. The receive routine from the Bus component calls this routine with a Generated etl::message but In This layer I dont know the type I just know the ID from the raw data. How is it possible to come from the raw ID number to the type and set it on the bus?

My thoughts are coming from this repo: 

https://github.com/intel/compile-time-init-build

**Comment by jwellbelove** [comment]
Sorry for the delay in replying, I don't get notifications for new 'discussions'.
Do you still need help with this?

**Reply by tunguskar** [reply]
I think we can close this I will come up with a more specific question
